sniproxy.cpp

```diff
@@ -190,7 +190,40 @@ SNIProxy::~SNIProxy()
 
 void SNIProxy::update()
 {
-    m_pixmap = QPixmap::fromImage(getImageNonComposite());
+    QImage newImage = getImageNonComposite();
+
+    if (m_pixmap.isNull()) {
+        m_pixmap = QPixmap::fromImage(newImage);
+    }
+    else if (!newImage.isNull()) {
+        auto isTransparentImage = [](const QImage &image) -> bool {
```

Comment:
if you don't capture anything you may as well just make it a normal static method rather than a lambda